dependency/vault_pki.go

```diff
@@ -66,40 +64,68 @@ func (d *VaultPKIQuery) Fetch(clients *ClientSet, opts *QueryOptions,
 	default:
 	}
 
-	var encPEM []byte
-	var err error
-	if encPEM, err = os.ReadFile(d.filePath); err != nil || len(encPEM) == 0 {
-		// no need to write to file as it is the template dest
-		encPEM, err = d.fetchPEM(clients)
-	}
-	if err != nil {
-		return nil, nil, errors.Wrap(err, d.String())
-	}
+	needsRenewal := fmt.Errorf("needs renewal")
+	getPEM := func(renew bool) ([]byte, error) {
```

Comment:
nit: I think this could be moved to a method on `(d *VaultPKIQuery)` to simplify this function

Reply:
Style difference. IMO it read better embedded as it is all internal code that needs re-running and it reads better as an internal function that you reuse vs. a loop. And it makes it explicit where it is used.

Reply:
:+1: